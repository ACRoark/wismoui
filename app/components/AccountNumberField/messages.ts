import { defineMessages } from 'react-intl';

export const scope = 'wismo.components.account-number-field';

export default defineMessages({
  accountNumberFieldLabel: {
    id: `${scope}.account-number-field-label`,
    defaultMessage: 'Account Number',
  },
});
