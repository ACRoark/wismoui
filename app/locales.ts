export const DEFAULT_LOCALE = 'en';

const PROD_LOCALES = {
  en: 'English',
};

const TEST_LOCALES = {
  en: 'English',
  es: 'Spanish',
};

export const SUPPORTED_LOCALES = process.env.NODE_ENV === 'production' ? PROD_LOCALES : TEST_LOCALES;
