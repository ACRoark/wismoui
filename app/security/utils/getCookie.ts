const getCookie = (name: string): string =>
  document.cookie.split('; ').reduce((previousValue: string, currentValue: string): string => {
    const parts = currentValue.split('=');

    return parts[0] === name ? decodeURIComponent(parts[1]) : previousValue;
  }, '');

export default getCookie;
