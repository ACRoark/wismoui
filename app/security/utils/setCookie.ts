interface ICookieOptions {
  days: number;
  path: string;
}

const setCookie = (name: string, value: string, options: ICookieOptions): void => {
  const expires = new Date(Date.now() + options.days * 864e5).toUTCString();

  document.cookie = `${name}=${encodeURIComponent(value)}; expires=${expires}; path=${options.path}`;
};

export default setCookie;
