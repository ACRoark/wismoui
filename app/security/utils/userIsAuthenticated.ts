import useAuth from 'hooks/useAuth';

const userIsAuthenticated = (): boolean => {
  const { isAuthenticated } = useAuth();

  return isAuthenticated;
};

export default userIsAuthenticated;
