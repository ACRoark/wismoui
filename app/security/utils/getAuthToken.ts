import getCookie from './getCookie';

const getAuthToken = (): string => getCookie('token');

export default getAuthToken;
