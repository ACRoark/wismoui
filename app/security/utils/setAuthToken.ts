import setCookie from './setCookie';

const setAuthToken = (value?: string): void => {
  if (value) {
    setCookie('token', `Bearer ${value}`, { days: 1, path: '/' });
  }
};

export default setAuthToken;
