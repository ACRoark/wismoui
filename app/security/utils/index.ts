import getAuthToken from './getAuthToken';
import setAuthToken from './setAuthToken';
import userIsAuthenticated from './userIsAuthenticated';

export { getAuthToken, setAuthToken, userIsAuthenticated };
