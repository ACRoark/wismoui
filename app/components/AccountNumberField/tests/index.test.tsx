import React from 'react';

import { SUPPORTED_LOCALES } from 'locales';

import AccountNumberField from 'components/AccountNumberField';
import { createSnapshotWithIntl, renderWithIntl } from 'testing/utils';

const accountNumber = '111233455567';

describe('<AccountNumberField />', (): void => {
  const languages = Object.keys(SUPPORTED_LOCALES);

  languages.forEach((language: string): void => {
    it(`should not log errors in console when rendered in ${language}`, (): void => {
      const spy = jest.spyOn(global.console, 'error');

      renderWithIntl(<AccountNumberField accountNumber={accountNumber}/>, language);

      expect(spy).not.toHaveBeenCalled();
    });

    it(`should render correctly when a valid account number is passed in and the current language is ${language}`, (): void => {
      const tree = createSnapshotWithIntl(<AccountNumberField accountNumber={accountNumber}/>, language);

      expect(tree).toMatchSnapshot();
    });
  });
});
