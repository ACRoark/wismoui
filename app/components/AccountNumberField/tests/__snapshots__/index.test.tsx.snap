// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AccountNumberField /> should render correctly when a valid account number is passed in and the current language is en 1`] = `
<div
  className="dte-wismo-base-field-section"
>
  <label
    className="dte-wismo-base-field-label"
  >
    Account Number
  </label>
  <span
    className="dte-wismo-base-field-children"
  >
    <span>
      1112 334 5556 7
    </span>
  </span>
</div>
`;

exports[`<AccountNumberField /> should render correctly when a valid account number is passed in and the current language is es 1`] = `
<div
  className="dte-wismo-base-field-section"
>
  <label
    className="dte-wismo-base-field-label"
  >
    NÃºmero de cuenta
  </label>
  <span
    className="dte-wismo-base-field-children"
  >
    <span>
      1112 334 5556 7
    </span>
  </span>
</div>
`;
